wagtail/models/__init__.py

```diff
@@ -1299,6 +1299,143 @@ def __init__(self, *args, **kwargs):
+                      increasing the path.max_length property \
+                    and UPDATE your database"
+                )
+            self.numchild += 1
+            node._cached_parent_obj = self
+
+        return
+
+    def _process_unordered_children(self, nodes):
+        """
+        Create child nodes without any specific order.
+        """
+        last_child = self.get_last_child()
+        max_length = self.__class__._meta.get_field("path").max_length
+        for node in nodes:
+            node.depth = self.depth + 1
+            node.path = last_child._inc_path()
+            last_child = node
+            if len(node.path) > max_length:
+                raise ValidationError(
+                    "The new node is too deep in the tree, try \
+                      increasing the path.max_length property \
+                    and UPDATE your database"
+                )
+            self.numchild += 1
+            node._cached_parent_obj = self
+
+        return
+
+    def _process_child_nodes(self, nodes):
+        """
+        Process the child nodes of this page, setting their paths and depths
+        as appropriate and also updating the parent's numchild. This is performed by the treebeard library when saving
+        the page to the database using their inbuilt tree functionality.
+        """
+
+        if self.is_leaf():
+            return self._process_leaf(nodes)
+
+        return self._process_unordered_children(nodes)
+
+    def _check_unique(self, children):
+        # Extract slugs from the children
+        # Children is a list of Page objects
+        slugs = [
+            child.slug for child in children if hasattr(child, "slug") and child.slug
+        ]
+        # Add the slugs of the current children
+        slugs.extend(self.get_children().values_list("slug", flat=True))
+
+        if len(slugs) != len(set(slugs)):
+            raise ValidationError(
+                {
+                    "slug": _(
+                        "Duplicate slugs in use within the parent page at '%(parent_url_path)s'"
+                    )
+                    % {
+                        "parent_url_path": self.path,
+                    }
+                }
+            )
+        for child in children:
+            if not hasattr(child, "slug") or not child.slug:
+                candidate_slug = slugify(child.title, allow_unicode=True)
+                suffix = 1
+                while candidate_slug in slugs:
+                    suffix += 1
+                    candidate_slug = "%s-%d" % (candidate_slug, suffix)
+                child.slug = candidate_slug
+                slugs.append(candidate_slug)
+
+            if child.locale_id is None:
+                child.locale = self.locale
+        return
+
+    def _save_to_db(self, children):
+        """
+        Load the pages into the database and save the parent page in the database
+        """
+
+        pages = Page.objects.bulk_create(children)
```

Comment:
Using `Page.objects.bulk_create()` here suggests we only want to create generic Page objects, but that is not the case is it? The type of each child should determine which manager is used here.